shopfloor/actions/barcode_parser.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2025 ACSONE SA/NV (https://www.acsone.eu)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo.addons.component.core import Component
+
+from ..actions.search import SearchAction
+
+
+class BarcodeResult:
+
+    __slots__ = ("type", "value", "raw")
+
+    def __init__(self, **kw) -> None:
+        for k in self.__slots__:
+            setattr(self, k, kw.get(k))
+
+
+class BarcodeParser(Component):
+    """
+    Some barcodes can have complex data structure
+    """
+
+    _name = "shopfloor.barcode.parser"
+    _inherit = "shopfloor.process.action"
+    _usage = "barcode"
```

Comment:
maybe I forgot how I implemented this machinery :sweat_smile:  but.. how is this supposed to kick in?

Reply:
That is in the second commit I lost but recovered.